src/main/filesystem.ts

```diff
@@ -83,9 +88,52 @@ export class KangarooFileSystem {
   }
 
   randomPasswordExists() {
-    const pwPath = path.join(this.appDataDir, '.pw');
+    const pwPath = path.join(this.profileDataDir, '.pw');
     return fs.existsSync(pwPath);
   }
+
+  async openLogs() {
+    try {
+      await shell.openPath(this.profileLogsDir);
+    } catch (e) {
+      dialog.showErrorBox('Failed to open logs folder', (e as any).toString());
+    }
+  }
+
+  async exportLogs() {
+    try {
+      const zip = new AdmZip();
+      zip.addLocalFolder(this.profileLogsDir);
+      const exportToPathResponse = await dialog.showSaveDialog({
+        title: 'Export Logs',
+        buttonLabel: 'Export',
+        defaultPath: `Moss_${app.getVersion()}_logs_${Date.now()}.zip`,
+      });
+      if (exportToPathResponse.filePath) {
+        zip.writeZip(exportToPathResponse.filePath);
+        shell.showItemInFolder(exportToPathResponse.filePath);
+      }
+    } catch (e) {
+      dialog.showErrorBox('Failed to export logs', (e as any).toString());
+    }
+  }
+
+  async factoryReset(keepLogs = false) {
+    if (keepLogs) throw new Error('Keeping logs across factory reset is currently not supported.');
+    if (platform.isWindows) {
```

Comment:
Windows needs special treatment, because it tends to keep files "in use" that we'd want to delete and then rejects deletion. And since factory reset needs to be triggered by the app, we can apparently not let loose of all files without exiting the app but then we have no process left to trigger the deletion anymore. It's a chicken-egg problem and only a problem on Windows, so we also delete browser data here via the electron API instead of just removing the containing folder.

Reply:
https://github.com/holochain/kangaroo-electron/pull/45/commits/a2adebc5f4b48d1d9de0cff1e444eb1870b03abb